src/Illuminate/Foundation/Application.php

```diff
@@ -530,6 +536,15 @@ protected function markAsRegistered($provider)
 		$this->serviceProviders[] = $provider;
 
 		$this->loadedProviders[$class] = true;
+		
+		$aliases = class_parents($class) + class_implements($class) + array($class);
+		foreach ($aliases as $alias)
+		{
+			if ( ! isset($this->registeredProviders[$alias]))
+			{
+				$this->registeredProviders[$alias] = $provider;
```

Comment:
I suppose.  Might look something more like this:

```php
$aliases = class_parents($class) + class_implements($class) + array($class);
$map = array_combine($aliases, array_fill(0, count($aliases), $provider));
$this->registeredProviders += $map;
```

Do note that `class_implements()` and etc. return `[A] => A` style arrays, so `array_merge()` is not necessary.  Doesn't matter, though.

However, this looks a bit more arcane, and I didn't see any real performance hit from the foreach.  I don't mind either way, I can change it to something else.